icu4c/source/test/intltest/measfmttest.cpp

```diff
@@ -3722,6 +3724,96 @@ void MeasureFormatTest::TestInvalidIdentifiers() {
     }
 }
 
+void MeasureFormatTest::TestIdentifierDetails() {
+    IcuTestErrorCode status(*this, "TestIdentifierDetails()");
+
+    MeasureUnit joule = MeasureUnit::forIdentifier("joule", status);
+    status.assertSuccess();
+    assertEquals("Initial joule", "joule", joule.getIdentifier());
+
+    MeasureUnit unit = joule.withPrefix(UMEASURE_SI_PREFIX_HECTO, status);
+    status.assertSuccess();
+    assertEquals("foo identifier", "hectojoule", unit.getIdentifier());
+
+    unit = unit.withPrefix(UMEASURE_BIN_PREFIX_EXBI, status);
+    status.assertSuccess();
+    assertEquals("foo identifier", "exbijoule", unit.getIdentifier());
+
+    // FIXME: test and implement desireable sorting behaviour.
```

Comment:
Desired sort order to be specified: [CLDR-14305](https://unicode-org.atlassian.net/browse/CLDR-14305)